query-engine/driver-adapters/js/connector-test-kit-executor/src/index.ts

```diff
@@ -63,7 +66,22 @@ async function handleRequest(method: string, params: unknown): Promise<unknown>
 
             console.error("Got `query`", params)
             const castParams = params as QueryPayload;
-            const result = await schemas[castParams.schemaId].query(JSON.stringify(castParams.query), "", castParams.txId)
+            const engine = schemas[castParams.schemaId]
+            const result = await engine.query(JSON.stringify(castParams.query), "", castParams.txId)
+
+            const parsedResult = JSON.parse(result)
+            if (parsedResult.errors) {
+                const error = parsedResult.errors[0]?.user_facing_error
+                if (error.error_code === 'P2036') {
+                    const jsError = adapters[castParams.schemaId].errorRegistry.consumeError(error.meta.id)
+                    if (!jsError) {
+                        throw new Error(`Something went wrong. Engine reported external error with id ${error.meta.id}, but it was not registered.`)
+                    } else {
+                        console.error("[nodejs] got error response from the engine caused by the driver: ", jsError)
```

Comment:
Should we just rethrow JS error here instead? 
In that case, `catch` inside of a main function will handle it and repond to the rust test suite the same way it reponds to JS error within executor itself and the error message & stacktrace would be visible within Rust test without extra logging.

Reply:
I see more tests failing due to rethrowing there, which is not to say we shouldn't do it. We should. I softened the else clause to log the error instead to isolate the causes of the errors to those attributed to the driver itself and not the adapter code with that wraps the error registry. Could you maybe take a look by changing this to throw and see why sometimes errors cannot be correlated?